src/cc65/declare.c

```diff
@@ -2079,8 +2089,14 @@ static unsigned ParseStructInit (Type* T, int AllowFlexibleMembers)
 
             /* Account for the data and output it if we have a full word */
             SI.ValBits += Entry->V.B.BitWidth;
-            CHECK (SI.ValBits <= INT_BITS);
-            if (SI.ValBits == INT_BITS) {
+            /* Make sure unsigned is big enough to hold the value, 30 bits.
+            ** This is 30 and not 32 bits because a 16-bit bit-field will
+            ** always be byte aligned, so will have padding before it.
+            ** 15 bits twice is the most we can have.
+            */
+            CHECK (SI.ValBits <= CHAR_BIT * sizeof(SI.BitVal));
```

Comment:
`CHAR_BIT` should be plural: `CHAR_BITS`.

Reply:
This `CHECK` is to make sure that we didn't overflow `BitVal` on the host platform. It's the host platform's `sizeof(BitVal)` and `CHAR_BIT` that are relevant here.

Reply:
> This `CHECK` is to make sure that we didn't overflow `BitVal` on the host platform. It's the host platform's `sizeof(BitVal)` and `CHAR_BIT` that are relevant here.

Ah, I see.

Reply:
It seem safe to consider this resolved...